### PR TITLE
Remove custom builds mako

### PR DESCRIPTION
**Pull request by guerler**
This PR replaces the custom build mako with a backbone view and corresponding API endpoints.

dbbuild

### CODE REVIEW

**Comment by bgruening** [comment]
Thanks @guerler for working on this!

**Comment by guerler** [comment]
Thanks @bgruening. I have not used this feature very often and would appreciate if someone could help me test these changes. ping @jennaj

**Comment by jennaj** [comment]
@guerler I can test, feedback in a bit

**Comment by jennaj** [comment]
Works great and updated styling is nice. 

All functions on the GC form itself worked (and quickly)

Function choices: Fa from history, Len pasted, and Len uploaded all create proper CGs
- listed correctly on the CG form, edit attributes genome list, upload genome list (therefore can be assigned as database metadata)
- can be used with trackster for visualization

The prior link for "Show loaded, system-installed builds" is no longer there, but it was a function that needed some tunings to make the info really useful (searchable list, instead of raw text blob). Maybe was removed on purpose?

Let me know if you need more done. Thanks!

**Comment by guerler** [comment]
Thanks a lot for the quick review @jennaj. The 'system-installed builds' are shown in a searchable list if any are installed otherwise the search field is hidden.

File: lib/galaxy/webapps/galaxy/api/users.py [hunk 1]
```diff
@@ -676,6 +679,127 @@ def set_communication(self, trans, id, payload={}, **kwd):
         trans.sa_session.flush()
         return {'message': message}
 
+    @expose_api
+    def get_custom_builds_metadata(self, trans, id, payload={}, **kwd):
+        """ Returns meta data for custom builds. """
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/users.py]: Can you please add some documentation to the new API calls (like the one `create()`, `update()` and `delete()` methods have)?

Comment (guerler) [reply lib/galaxy/webapps/galaxy/api/users.py]: Yes, of course, thanks for pointing it out @nsoranzo

**Comment by dannon** [comment]
(talking through API changes with @guerler offline, review in progress)

**Comment by dannon** [comment]
Thanks for making those API adjustments for me, looks good now.

**Comment by nsoranzo** [comment]
And also thanks for the API docs @guerler!

**Pull request merged**